### PR TITLE
CB-10819 Adding snapshotting and improving build system

### PR DESCRIPTION
**Pull request by dblotsky**
- Added 'snap' command to Makefile and gulpfile. 
- Fixed 'newversion' command.
- Made fetched files live under /reference/. 
- Refactored and removed incrementversion.js in favour of simple FS commands.

### CODE REVIEW

**Comment by dblotsky** [comment]
@riknoll @rakatyal please review when you have a moment.

**Comment by dblotsky** [comment]
@sarangan12 @nikhilkh please review when you have a moment.

**Comment by nikhilkh** [comment]
You'll need re-directs for all the links that you changed. Also, the docs have lots of references to them and they need to be changed.

**Comment by dblotsky** [comment]
@nikhilkh good catch! Updated internal links and added redirects. Please give it another look.

File: www/docs/en/dev/guide/hybrid/plugins/index.md [hunk 1]
```diff
@@ -44,13 +44,12 @@ For each corresponding native interface, see the list at the end of
...
 
 In addition to these instructions, when preparing to write a plugin it
-is best to look over
-[existing plugins](http://cordova.apache.org/contribute)
+is best to look over [existing plugins](http://cordova.apache.org/contribute)
 for guidance.
 
 ## Building a Plugin
 
-Application developers use the CLI's [plugin add command](../../../cordova-cli/index.html#cordova-plugin-add-command) to add a plugin to a project. The
```
Comment (dblotsky) [inline www/docs/en/dev/guide/hybrid/plugins/index.md]: @nikhilkh this heading doesn't exist in the master cordova-cli reference. I assumed you meant `cordova-plugin-command`. Is that correct?

Comment (nikhilkh) [inline www/docs/en/dev/guide/hybrid/plugins/index.md]: Good catch - yes, that's a bug!

**Comment by nikhilkh** [comment]
LGTM

**Pull request merged**